packages/core/content-manager/server/src/controllers/collection-types.ts

```diff
@@ -1,9 +1,112 @@
-import { setCreatorFields, mapAsync, pipeAsync } from '@strapi/utils';
+import { setCreatorFields, mapAsync, pipeAsync, errors } from '@strapi/utils';
 import { getService } from '../utils';
 import { validateBulkActionInput } from './validation';
 import { getProhibitedCloningFields, excludeNotCreatableFields } from './utils/clone';
 import { getDocumentDimensions } from './utils/dimensions';
 
+/**
+ * Create a new document.
+ *
+ * @param ctx - Koa context
+ * @param opts - Options
+ * @param opts.populate - Populate options of the returned document. By default the entityManager will populate all relations.
+ */
+const createDocument = async (ctx: any, opts?: { populate?: object }) => {
+  const { userAbility, user } = ctx.state;
+  const { model } = ctx.params;
+  const { body } = ctx.request;
+
+  const entityManager = getService('entity-manager');
+  const permissionChecker = getService('permission-checker').create({ userAbility, model });
+
+  if (permissionChecker.cannot.create()) {
+    throw new errors.ForbiddenError();
+  }
+
+  const pickPermittedFields = permissionChecker.sanitizeCreateInput;
+  const setCreator = setCreatorFields({ user });
+  const sanitizeFn = pipeAsync(pickPermittedFields, setCreator as any);
+  const sanitizedBody = await sanitizeFn(body);
+
+  const { locale, status = 'draft' } = getDocumentDimensions(body);
+
+  return entityManager.create(model, {
+    data: sanitizedBody,
+    locale,
+    status,
+    populate: opts?.populate,
+  });
+
+  // TODO: Revert the creation if create permission conditions are not met
```

Comment:
is this not covered in L22?

Reply:
it's a bit different, on L22 we only check if the user has CRUD permissions to create, but we don't acount for custom conditions are granular at the field level, for that it's necessary a created document. This has been in TODO  for a long time, so I didnt want to add more complexity into the PR